src/app/clusters/general-diagnostics-server/GeneralDiagnosticsCluster.cpp

```diff
@@ -274,6 +275,24 @@ DataModel::ActionReturnStatus GeneralDiagnosticsCluster::ReadAttribute(const Dat
     case GeneralDiagnostics::Attributes::TestEventTriggersEnabled::Id: {
         bool isTestEventTriggersEnabled = IsTestEventTriggerEnabled();
         return encoder.Encode(isTestEventTriggersEnabled);
+    }
+    case GeneralDiagnostics::Attributes::DeviceLoadStatus::Id: {
+        InteractionModelEngine * interactionModel            = InteractionModelEngine::GetInstance();
+        SessionManager * sessionManager                      = interactionModel->GetExchangeManager()->GetSessionManager();
```

Comment:
This way of implementing it is probably not unit testable.

Do we have a way to make this unit testable (maybe having this data via some provider interface ... where we can have a IME provider to receive it)?

Otherwise we are generally missing unit tests here. Do we claim "code is simple enough" ?